config/tests/multi-pipe-ringbuf.yaml

```diff
@@ -0,0 +1,115 @@
+##########################################
+#
+##########################################
+---
+type: config
+# Logging level can be one of:
+# OFF, ERROR, WARN, INFO, DEBUG, DEBUG2 (case insensitive)
+# Note DEBUG and DEBUG2 require a build with (-DCMAKE_BUILD_TYPE=Debug)
+log_level: debug
+#    log_level: info
+cpu_affinity: [2,3,4,5,8,9,10,11]
+frame_arrival_period: 0.001
+
+buffer_depth: 4
+
+# data gen:
+num_frames: 20
+
+input_size: 4
+output_size: 5
+
+ring_buffer_size: 10
+
+# Pool
+main_pool:
+    kotekan_metadata_pool: chordMetadata
+    num_metadata_objects: 15 * buffer_depth
+
+# Buffers
+host_in_buffer:
+    kotekan_buffer: standard
+    num_frames: buffer_depth
+    frame_size: input_size
+    metadata_pool: main_pool
+
+# Buffer used for signalling only, between the two GPU pipelines.
+host_signal_buffer:
+    kotekan_buffer: ring
+    #ring_buffer_size: 
+    metadata_pool: main_pool
+
+host_out_buffer:
+    kotekan_buffer: standard
+    num_frames: buffer_depth
+    frame_size: output_size
+    metadata_pool: main_pool
+
+gen_data:
+    kotekan_stage: testDataGen
+    type: const8
+    values: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10] #, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]
+    #    array_shape: [64, 16]
+    #dim_name: ["X", "Y"]
+    out_buf: host_in_buffer
+
+gpuA:
+    #profiling: true
+    #log_profiling: true
+    kernel_path: "../../lib/cuda/kernels/"
+    commands: &command_list
+    - name: cudaInputData
+      in_buf: host_in
+      gpu_mem: in
+    - name: cudaSyncInput
+    - name: cudaCopyToRingbuffer
```

Comment:
Yeah, we're going to have to be careful about that.  Kendrick wrote a stand-alone kernel that duplicates the functionality of cudaMemcpyAsync but operates without the copy engine - we may need to use that in some places.  I'm not going to worry too much about that here, since cudaCopy{To,From}Ringbuffer are really just demos of how to do the copies, and usually they would be integrated into a kernel wrapper directly, and then not require a(n async) copy.